### ISSUE TITLE
Query: compilation error for queries with join where the inner key is navigation

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#11245]
repro using GearsOfWar model

```cs
from g1 in ctx.Gears
join g2 in ctx.Gears on g1.AssignedCity equals g2.AssignedCity
select new { Nickname1 = g1.Nickname, Nickname2 = g2.Nickname };
```

throws:

```
Property 'Microsoft.EntityFrameworkCore.TestModels.GearsOfWarModel.City AssignedCity' is not defined for type 'Microsoft.EntityFrameworkCore.TestModels.GearsOfWarModel.City'
	at System.Linq.Expressions.Expression.Property(Expression expression, PropertyInfo property)
	at System.Linq.Expressions.Expression.MakeMemberAccess(Expression expression, MemberInfo member)
	at Microsoft.EntityFrameworkCore.Internal.ExpressionExtensions.MakeMemberAccess(Expression expression, MemberInfo member)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.<>c__DisplayClass19_0.<VisitMember>b__1(Expression e)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.CreateSubqueryForNavigations(Expression outerQuerySourceReferenceExpression, ICollection`1 navigations, Func`2 propertyCreator)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.RewriteNavigationProperties(IReadOnlyList`1 properties, IQuerySource querySource, Expression expression, Expression declaringExpression, String propertyName, Type propertyType, Func`2 propertyCreator, Func`2 conditionalAccessPropertyCreator)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.<>c__DisplayClass19_0.<VisitMember>b__0(IReadOnlyList`1 ps, IQuerySource qs)
	at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.BindPropertyExpressionCore[TResult](Expression propertyExpression, IQuerySource querySource, Func`3 propertyBinder)
	at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.BindNavigationPathPropertyExpression[TResult](Expression propertyExpression, Func`3 propertyBinder)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.VisitMember(MemberExpression node)
	at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
	at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.NavigationRewritingQueryModelVisitor.VisitJoinClauseInternal(JoinClause joinClause)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.NavigationRewritingQueryModelVisitor.VisitJoinClause(JoinClause joinClause, QueryModel queryModel, Int32 index)
	at Remotion.Linq.Clauses.JoinClause.Accept(IQueryModelVisitor visitor, QueryModel queryModel, Int32 index)
	at Remotion.Linq.QueryModelVisitorBase.VisitBodyClauses(ObservableCollection`1 bodyClauses, QueryModel queryModel)
	at Remotion.Linq.QueryModelVisitorBase.VisitQueryModel(QueryModel queryModel)
	at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.NavigationRewritingExpressionVisitor.Rewrite(QueryModel queryModel, QueryModel parentQueryModel)
	at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.OptimizeQueryModel(QueryModel queryModel, Boolean asyncQuery)
	at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.OptimizeQueryModel(QueryModel queryModel, Boolean asyncQuery)
	at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel)
	at Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery[TResult](QueryModel queryModel)
	--- End of stack trace from previous location where exception was thrown ---
	at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
	at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](Expression query, IQueryModelGenerator queryModelGenerator, IDatabase database, IDiagnosticsLogger`1 logger, Type contextType)
	at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass13_0`1.<Execute>b__0()
	at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
	at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
	at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
	at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
	at Remotion.Linq.QueryableBase`1.GetEnumerator()
```

**Comment by maumar** [comment]
problem is that inner key nav rewrite goes thru a different code path (needs to be converted to subquery rather than join, because chicken-and-egg problem), and we have some invalid assumptions when the key doesn't have a scalar member at the end

**Comment by maumar** [comment]
fixed in fdb8eceb74a3fd585f90eced744977f74477b1dd
